mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -567,6 +567,8 @@ impl Nexus {
         }
 
         if r.await.unwrap() {
+            // Update the child states to remove them from the config file.
+            NexusChild::save_state_change();
```

Comment:
Forgot to ask last time around but why does this not take &self? i.e nexus.config_sync() would be more appropriate?

Reply:
Each call to save_state_change gets the state of every child for every nexus in the running system and updates the config file accordingly. IOW it overwrites the config file in its entirety rather than updating it for a specific nexus. That was why I didn't use _nexus.config_sync()_ as it implies we are only updating the config for that nexus.

The reason I went for this approach was that it mirrors what we do for the mayastor config.